piu/cli.py

```diff
@@ -247,6 +272,13 @@ def cli(ctx, config_file):
     envvar="PIU_CHECK_INSTANCE",
     default=True,
 )
+@click.option(
+    "-i",
+    "--ssh-public-key",
+    help="The public key to use to SSH",
+    type=click.Path(),
+    default=os.path.expanduser("~/.ssh/id_rsa.pub"),
```

Comment:
This is going to be very frustrating for the users (especially ones that use non-RSA keys). I'd change this to either pull the key from the users API, or get it from the agent (`ssh-add -L` lists all currently configured public keys).

Reply:
You can also support users with custom configs (i.e. run `ssh -G odd-eu-central-1-…` and parse it for `identityfile` sections), but I'm not sure if it's really worth it.

Reply:
As discussed offline the key is saved after the first use. If the key is not present at the default path then the user is prompted to enter the path to the key. So it should not be as inconvenient for users. This still leaves out the case where the key has not been loaded into the agent but the tool currently does not support this anyway.